src/couch/src/couch_att.erl

```diff
@@ -732,13 +735,19 @@ upgrade_encoding(true) -> gzip;
 upgrade_encoding(false) -> identity;
 upgrade_encoding(Encoding) -> Encoding.
 
-
 max_attachment_size() ->
-    case config:get("couchdb", "max_attachment_size", "infinity") of
+    max_attachment_size(config:get("couchdb", "max_attachment_size", ?GB)).
+
+max_attachment_size(MaxAttSizeConfig) ->
+    case MaxAttSizeConfig of
         "infinity" ->
             infinity;
+        MaxAttSize when is_list(MaxAttSize) ->
+            list_to_integer(MaxAttSize);
+        MaxAttSize when is_integer(MaxAttSize) ->
```

Comment:
You wouldn't need this clause if you change the define to be a string.
```
-define(GB, "1073741824").
```

Reply:
Considering https://github.com/apache/couchdb/pull/3347#discussion_r564547120 , I'd hope going for a type check has better performance than a conversion. What do you think?